### PR TITLE
🐛   Adjust records generated during view change so their date matches the view date

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Currently, when a new view is generated we get the current timestamp to define the view date.
Then we take a full snapshot, which consists [in 3-4 records](https://github.com/DataDog/browser-sdk/blob/c050a2006a2b0f9ed1175478abb90bc0cec7f250/packages/rum/src/domain/record/record.ts#L36-L70) every time a record is emitted, we get the current timestamp to define its date

This causes an issue during replay, when seeking: when the user clicks on a View event, the player actually shows the previous view, because the records corresponding to the new view are a few milliseconds after the view start.

This PR set the view date to the fullsnapshot emitted records when a view change

## Changes

Set the view date to the fullsnapshot emitted records when a view change

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum/src/boot/startRecording.spec.ts [hunk 1]
```diff
@@ -165,6 +165,37 @@ describe('startRecording', () => {
...
+          { type: RecordType.Focus, timestamp: Date.now(), data: jasmine.any(Object) },
+          { type: RecordType.FullSnapshot, timestamp: Date.now(), data: jasmine.any(Object) },
+          { type: RecordType.VisualViewport, timestamp: Date.now(), data: jasmine.any(Object) },
+          { type: RecordType.ViewEnd, timestamp: Date.now() },
+        ])
+        clock.cleanup()
+
+        readRequestSegment(calls.mostRecent(), (segment) => {
+          expect(segment.records).toEqual([
+            { type: RecordType.Meta, timestamp: 1, data: jasmine.any(Object) },
```
Comment (BenoitZugmeyer) [inline packages/rum/src/boot/startRecording.spec.ts:188]: **💬 suggestion:** ‏Use a constant like `VIEW_TIMESTAMP = 1` to be more explicit

File: packages/rum/src/types.ts [hunk 2]
```diff
@@ -55,6 +50,7 @@ export type RecordType = typeof RecordType[keyof typeof RecordType]
 
 export interface FullSnapshotRecord {
   type: typeof RecordType.FullSnapshot
+  timestamp: number
```
[the remark this answers is not in the document]
Comment (amortemousque) [reply packages/rum/src/types.ts:53]: renaming `timestamp` to `timeStamp` wont it be a breaking change?

File: packages/rum/src/boot/startRecording.spec.ts (hunk 1 above)
Comment (amortemousque) [reply packages/rum/src/boot/startRecording.spec.ts:188]: sure

File: packages/rum/src/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenoitZugmeyer) [reply packages/rum/src/types.ts:53]: Ah, right. Let's just change the type them

**Pull request merged**